notebooks/data_obfuscation.ipynb

```diff
@@ -0,0 +1,1222 @@
+{
```

Comment:
<p>If Argon2 was found to be the most advanced/preferred technique, should we use that for encrypting the below example then? </p><p> </p><p>If this is just to demonstrate how we can use any technique to encrypt a dataset of this format, then its fine to just use sha256.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/operate-first-data-science-community/pull/33/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Reply:
The notebook serves the purpose of giving a brief intro about different obfuscation methods, from weaker to stronger once. 

In our present case, we will be obfuscating first name and last name only. We are not obfuscating the emails, instead we are splitting it and only keeping the domain part of it. As I understand, we want the hashed version for first name and last name to be same in all the datasets we fetch, so that we can compare and see if a person attending meetup 1 is present in meetup 2. Using [Argon2](https://en.wikipedia.org/wiki/Argon2), I believe it gives us a different hash value for different entry, even if the word is similar (because of the [salt](https://en.wikipedia.org/wiki/Salt_(cryptography))). Its application lies mostly in storing hashed password in the database, such that even if two different users stores the same password. Then the hashed version of them are different. In our present scenario, I believe hashing with SHA-256 will do the work.

Reply:
<p>For our purposes, would it make sense to combine first and last name into a single string before hashing, so we just have one unique identifier value for each attendee? Does combining these two fields increase or decrease security? </p>

Reply:
:+1:

Reply:
@oindrillac @MichaelClifford 
Updated the notebook: 
 - merged the first name and last name as one
 - applied double hashing technique by hashing the name with sha-256 and later hashing the hashed version with sha-512.  Thank you for your feedback. :)